test/unit/uk/gov/hmrc/apinotificationqueue/service/WarningEmailPollingServiceSpec.scala

```diff
@@ -0,0 +1,95 @@
+
+class WarningEmailPollingServiceSpec extends UnitSpec with MockitoSugar with Eventually {
+
+  trait Setup {
+
+    val mockNotificationRepository = mock[NotificationRepository]
+    val mockEmailConnector = mock[EmailConnector]
+    val mockServiceConfiguration = mock[ServiceConfiguration]
+    val testActorSystem = ActorSystem("WarningEmailPollingService")
+
+    val year = 2017
+    val monthOfYear = 7
+    val dayOfMonth = 4
+    val hourOfDay = 13
+    val minuteOfHour = 45
+    val timeReceived = new DateTime(year, monthOfYear, dayOfMonth, hourOfDay, minuteOfHour)
+    val latestReceived = timeReceived.plus(1)
+    val clientId1 = "clientId1"
+    val clientOverThreshold1 = ClientOverThreshold(clientId1, 2, timeReceived, latestReceived)
+
+    val sendEmailRequest = SendEmailRequest(List(Email("some-email@address.com")),
+      "customs_pull_notifications_warning",
+      Map("clientId_0" -> "some-client-id",
+        "notificationTotal_0" -> "some-total",
+        "oldestNotification_0" -> "some-date",
+        "latestNotification_0" -> "some-date",
+        "queueThreshold" -> "some-threshold"),
+      force = false)
+  }
+
+  "WarningEmailService" should {
+    "send an email" in new Setup {
+      when(mockServiceConfiguration.getInt("notification.email.queueThreshold")).thenReturn(2)
+      when(mockServiceConfiguration.getString("notification.email.address")).thenReturn("some-email@address.com")
+      when(mockNotificationRepository.fetchOverThreshold(2)).thenReturn(Future.successful(List(clientOverThreshold1)))
+      when(mockServiceConfiguration.getInt("notification.email.interval")).thenReturn(1)
+
+      val warningEmailService = new WarningEmailPollingService(mockNotificationRepository, mockEmailConnector, testActorSystem, mockServiceConfiguration)
+      Thread.sleep(1000)
+      val emailRequestCaptor: ArgumentCaptor[SendEmailRequest] = ArgumentCaptor.forClass(classOf[SendEmailRequest])
+      verify(mockEmailConnector).send(emailRequestCaptor.capture())
+
+      val request: SendEmailRequest = emailRequestCaptor.getValue
+      request.to.head.value shouldBe "some-email@address.com"
+      request.templateId shouldBe "customs_pull_notifications_warning"
+      ((request.parameters.keySet -- sendEmailRequest.parameters.keySet)
+        ++ (sendEmailRequest.parameters.keySet -- request.parameters.keySet)).size shouldBe 0
+    }
+
+    "no email should be sent when no clients breach queue threshold" in new Setup {
+      when(mockServiceConfiguration.getInt("notification.email.queueThreshold")).thenReturn(200)
+      when(mockServiceConfiguration.getString("notification.email.address")).thenReturn("some-email@address.com")
+      when(mockNotificationRepository.fetchOverThreshold(200)).thenReturn(Future.successful(List.empty))
+
+      val warningEmailService = new WarningEmailPollingService(mockNotificationRepository, mockEmailConnector, testActorSystem, mockServiceConfiguration)
+      testActorSystem.terminate()
```

Comment:
is there any reason why ` testActorSystem.terminate()` is here and not in other tests?

Reply:
It appeared that the test didn't clean up immediately so to stop the logger continuing I killed the ActorSystem as part of the test. The result was that log statements stopped immediately.